python/ec2_sg_public_ingress_remove.py

```diff
@@ -0,0 +1,165 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Ensure that no security groups allow public access to the specified ports.
+# Description: Checks that all security groups block access to the specified ports and removes ip permissions that violate the rule.
+#
+# Trigger Type: Change Triggered
+# Scope of Changes: EC2:SecurityGroup
+# Accepted Parameters: examplePort1, exampleRange1, examplePort2, ...
+# Example Value: 8080, 1-1024, 2375, ...
+
+
+import json
+import boto3
+import botocore
+
+
+
+APPLICABLE_RESOURCES = ["AWS::EC2::SecurityGroup"]
+
+def capitalize(x):
+   if isinstance(x, list):
+     return [capitalize(v) for v in x]
+   elif isinstance(x, dict):
+     return {k[0].upper() + k[1:]: capitalize(v) for k, v in x.items()}
+   else:
+     return x
+
+def expand_range(ports):
+    if "-" in ports:
+        return range(int(ports.split("-")[0]), int(ports.split("-")[1])+1)
+    else:
+        return [int(ports)]
+
+def revoke_permissions(permissions):
+    #global group_id
+    group_id = configuration_item["configuration"]["groupId"]
+    client = boto3.client("ec2");
+    print("revoking permissions in ", group_id)
+    #Capatilize the first letter of each key
+    revoke_permissions = [capitalize(permission)]
+    #Transform Ipv4Ranges -> IpRanges
+    for element in revoke_permissions: 
+        #dictionary[new_key] = dictionary.pop(old_key)
```

Comment:
I don't think this comment is needed

Reply:
yes 
 "#dictionary[new_key] = dictionary.pop(old_key)"
 can be removed